akka-distributed-data/src/main/scala/akka/cluster/ddata/DurableStore.scala

```diff
@@ -0,0 +1,257 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.cluster.ddata
+
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+import scala.collection.JavaConverters._
+import scala.collection.breakOut
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+import akka.actor.Actor
+import akka.actor.ActorLogging
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.DeadLetterSuppression
+import akka.actor.ExtendedActorSystem
+import akka.actor.Props
+import akka.cluster.Cluster
+import akka.cluster.ddata.Replicator.Internal.DataEnvelope
+import akka.serialization.Serialization
+import akka.serialization.SerializationExtension
+import akka.serialization.SerializerWithStringManifest
+import com.typesafe.config.Config
+import org.mapdb.DBMaker
+import org.mapdb.DataIO
+import org.mapdb.DataInput2
+import org.mapdb.DataOutput2
+import org.mapdb.HTreeMap
+import org.mapdb.Serializer
+
+/**
+ * An actor implementing the durable store for the Distributed Data `Replicator`
+ * have to implement the protocol with the messages defined here.
+ *
+ * At startup the `Replicator` creates the durable store actor and sends the
+ * `Load` message to it. It must then reply with 0 or more `LoadData` messages
+ * followed by one `LoadCompleted` message to the `sender` (the `Replicator`).
+ *
+ * If the `Load` fails it can throw `LoadFailed` and the `Replicator` supervisor
+ * will stop itself and the durable store.
+ *
+ * When the `Replicator` needs to store a value it sends a `Store` message
+ * to the durable store actor, which must then reply with the `successMsg` or
+ * `failureMsg` to the `replyTo`.
+ */
+object DurableStore {
+  final case class Store(key: String, data: ReplicatedData, reply: Option[StoreReply])
```

Comment:
Please comment the relation of these two classes:
 - Store contains a StoreReply (optionally), which defines an actor and the messages it should receive on success or failure.